packages/connector-apple/src/index.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  ConnectorMetadata,
+  GetAccessToken,
+  GetAuthorizationUri,
+  ValidateConfig,
+  GetUserInfo,
+  ConnectorError,
+  ConnectorErrorCodes,
+  SocialConnector,
+  GetConnectorConfig,
+} from '@logto/connector-types';
+import { conditional } from '@silverhand/essentials';
+import { createRemoteJWKSet, jwtVerify } from 'jose';
+
+import { scope, defaultMetadata, jwksUri, issuer } from './constant';
+import { appleConfigGuard, AppleConfig, appleDataGuard } from './types';
+
+// TO-DO: support nonce validation
+export default class AppleConnector implements SocialConnector<string> {
+  public metadata: ConnectorMetadata = defaultMetadata;
+
+  constructor(public readonly getConfig: GetConnectorConfig<AppleConfig>) {}
+
+  public validateConfig: ValidateConfig = async (config: unknown) => {
+    const result = appleConfigGuard.safeParse(config);
+
+    if (!result.success) {
+      throw new ConnectorError(ConnectorErrorCodes.InvalidConfig, result.error.message);
+    }
+  };
+
+  public getAuthorizationUri: GetAuthorizationUri = async (state, redirectUri) => {
+    const config = await this.getConfig(this.metadata.id);
+
+    const queryParameters = new URLSearchParams({
+      client_id: config.clientId,
+      redirect_uri: redirectUri,
+      scope,
```

Comment:
Since this `scope` is empty in the constant file, do we need it here?

Reply:
don't know if it's required, also we can keep this param to remind us to add more scopes in the future if needed

Reply:
don't know if it's required, also we can keep this param to remind us to add more scopes in the future if needed

Reply:
im ok to remove